### PR TITLE
Block cleaner should catch NonFatal exception

### PR DESCRIPTION
**Pull request by ulysses-you**
### _Why are the changes needed?_

The daemonset may encounter some pontential issue during long-time running. It should be robust to catch `NonFatal` exception with it.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by ulysses-you** [comment]
cc @yaooqinn @zwangsheng

File: tools/spark-block-cleaner/src/main/scala/org/apache/kyuubi/tools/KubernetesSparkBlockCleaner.scala [hunk 1]
```diff
@@ -185,8 +187,14 @@ object KubernetesSparkBlockCleaner extends Logging {
       val hasFinished = new CountDownLatch(cacheDirs.length)
       cacheDirs.foreach { dir =>
         threadPool.execute(() => {
-          doCleanJob(dir)
-          hasFinished.countDown()
+          try {
+            doCleanJob(dir)
+          } catch {
+            case NonFatal(e) =>
+              error("failed to clean", e)
```
Comment (yaooqinn) [inline tools/spark-block-cleaner/src/main/scala/org/apache/kyuubi/tools/KubernetesSparkBlockCleaner.scala:194]: print the dir too?

Comment (ulysses-you) [reply tools/spark-block-cleaner/src/main/scala/org/apache/kyuubi/tools/KubernetesSparkBlockCleaner.scala]: added

**Review by yaooqinn — APPROVED** [review]

**Comment by zwangsheng** [comment]
Can we figure out where cause this NonFatal Exception ？

**Comment by ulysses-you** [comment]
> Can we figure out where cause this NonFatal Exception ？

@zwangsheng I see a NPE error in `doClean` code and seems `blockManagerDir` become a null value. Not sure how this happened, but we need catch this exception.

**Pull request closed without merging**